src/web/error.rs

```diff
@@ -132,6 +136,110 @@ impl From<PoolError> for IronError {
     }
 }
 
+#[derive(Debug, thiserror::Error)]
+#[allow(dead_code)] // FIXME: remove after iron is gone
+pub enum AxumNope {
+    #[error("Requested resource not found")]
+    ResourceNotFound,
+    #[error("Requested build not found")]
+    BuildNotFound,
+    #[error("Requested crate not found")]
+    CrateNotFound,
+    #[error("Requested owner not found")]
+    OwnerNotFound,
+    #[error("Requested crate does not have specified version")]
+    VersionNotFound,
+    #[error("Search yielded no results")]
+    NoResults,
+    #[error("Internal server error")]
+    InternalServerError,
+    #[error("internal error")]
+    InternalError(#[from] anyhow::Error),
+    #[error("internal threading error")]
+    InternalTokio(#[from] tokio::task::JoinError),
+}
+
+impl IntoResponse for AxumNope {
+    fn into_response(self) -> AxumResponse {
+        match self {
+            AxumNope::ResourceNotFound => {
+                // user tried to navigate to a resource (doc page/file) that doesn't exist
+                AxumErrorPage {
+                    title: "Requested resource does not exist",
+                    message: Some("no such resource".into()),
+                    status: StatusCode::NOT_FOUND,
+                }
+                .into_response()
+            }
+
+            AxumNope::BuildNotFound => AxumErrorPage {
+                title: "The requested build does not exist",
+                message: Some("no such build".into()),
+                status: StatusCode::NOT_FOUND,
+            }
+            .into_response(),
+
+            AxumNope::CrateNotFound => {
+                // user tried to navigate to a crate that doesn't exist
+                // TODO: Display the attempted crate and a link to a search for said crate
```

Comment:
is this something that exists today, or a new feature we want to add? I don't want to merge code that introduces user-visible regressions (perf regressions and messy code are ok, but I don't want to regress features)

Reply:
are you referring to the error variant? or the `TODO` message? Both exist in the old `Nope` type too

Reply:
The TODO. Ok, if it's pre-existing that's fine.